src/test_application/OSPMagnum.cpp

```diff
@@ -160,76 +162,178 @@ osp::active::ActiveScene& OSPMagnum::scene_create(std::string && name)
+    {"L", {osp::sc_keyboard, (int)Key_t::L  }},
+    {"M", {osp::sc_keyboard, (int)Key_t::M  }},
+    {"N", {osp::sc_keyboard, (int)Key_t::N  }},
+    {"O", {osp::sc_keyboard, (int)Key_t::O  }},
+    {"P", {osp::sc_keyboard, (int)Key_t::P  }},
+    {"Q", {osp::sc_keyboard, (int)Key_t::Q  }},
+    {"R", {osp::sc_keyboard, (int)Key_t::R  }},
+    {"S", {osp::sc_keyboard, (int)Key_t::S  }},
+    {"T", {osp::sc_keyboard, (int)Key_t::T  }},
+    {"U", {osp::sc_keyboard, (int)Key_t::U  }},
+    {"V", {osp::sc_keyboard, (int)Key_t::V  }},
+    {"W", {osp::sc_keyboard, (int)Key_t::W  }},
+    {"X", {osp::sc_keyboard, (int)Key_t::X  }},
+    {"Y", {osp::sc_keyboard, (int)Key_t::Y  }},
+    {"Z", {osp::sc_keyboard, (int)Key_t::Z  }},
+
+    //Number keys
+    {"0", {osp::sc_keyboard, (int)Key_t::NumZero  }},
+    {"1", {osp::sc_keyboard, (int)Key_t::NumOne  }},
+    {"2", {osp::sc_keyboard, (int)Key_t::NumTwo  }},
+    {"3", {osp::sc_keyboard, (int)Key_t::NumThree  }},
+    {"4", {osp::sc_keyboard, (int)Key_t::NumFour  }},
+    {"5", {osp::sc_keyboard, (int)Key_t::NumFive  }},
+    {"6", {osp::sc_keyboard, (int)Key_t::NumSix  }},
+    {"7", {osp::sc_keyboard, (int)Key_t::NumSeven  }},
+    {"8", {osp::sc_keyboard, (int)Key_t::NumEight  }},
+    {"9", {osp::sc_keyboard, (int)Key_t::NumNine  }},
+
+    //Function keys
+    {"F1", {osp::sc_keyboard, (int)Key_t::F1  }},
+    {"F2", {osp::sc_keyboard, (int)Key_t::F2  }},
+    {"F3", {osp::sc_keyboard, (int)Key_t::F3  }},
+    {"F4", {osp::sc_keyboard, (int)Key_t::F4  }},
+    {"F5", {osp::sc_keyboard, (int)Key_t::F5  }},
+    {"F6", {osp::sc_keyboard, (int)Key_t::F6  }},
+    {"F7", {osp::sc_keyboard, (int)Key_t::F7  }},
+    {"F8", {osp::sc_keyboard, (int)Key_t::F8  }},
+    {"F9", {osp::sc_keyboard, (int)Key_t::F9  }},
+    {"F10", {osp::sc_keyboard, (int)Key_t::F10  }},
+    {"F11", {osp::sc_keyboard, (int)Key_t::F11  }},
+    {"F12", {osp::sc_keyboard, (int)Key_t::F12  }},
+
+    //Mouse
+    {"RMouse", {osp::sc_mouse, (int)OSPMagnum::MouseEvent::Button::Right }},
+    {"LMouse", {osp::sc_mouse, (int)OSPMagnum::MouseEvent::Button::Left }},
+    {"MMouse", {osp::sc_mouse, (int)OSPMagnum::MouseEvent::Button::Middle }}
+};
+
+std::vector<osp::ButtonVarConfig> parse_control(std::string_view str) {
+    //Split by +
+    constexpr std::string_view delim = "+";
+    std::vector<osp::ButtonVarConfig> handlers;
+
+    //If none, then no actions
+    if (str == "None") {
+        return handlers;
+    }
+
+    auto start = 0U;
+    auto end = str.find(delim);
+
+    //If no '+' is found
+    if (end == std::string_view::npos) {
+        try {
+            auto const& [device, button]
+                = buttonMap.at(str);
+
+            handlers.push_back(osp::ButtonVarConfig(device, button, VarTrig_t::PRESSED, false, VarOp_t::OR));
+        }
+        catch (std::out_of_range oor) {
+            std::cout << oor.what() << str << std::endl;
+        }
+    }
+    else {
+        while (end != std::string_view::npos)
```

Comment:
I think it's weird that you've got this while loop in the else-condition.

It seems to me, based on trying to understand the code, that you could have only the while loop, without needing to have the if/else branch.